src/compile.ts

```diff
@@ -101,42 +226,103 @@ export function compile(
       );
 
       function visitor(node: ts.Node): ts.Node | ts.Node[] {
-        const visit = () => {
-          if (checker.isAppsyncResolver(node)) {
-            return visitAppsyncResolver(node);
-          } else if (checker.isAppsyncField(node)) {
-            return visitAppsyncField(node);
-          } else if (checker.isNewStepFunction(node)) {
-            return visitStepFunction(node);
-          } else if (checker.isReflectFunction(node)) {
-            return errorBoundary(() =>
-              toFunction(NodeKind.FunctionDecl, node.arguments[0])
-            );
-          } else if (checker.isEventBusWhenFunction(node)) {
-            return visitEventBusWhen(node);
-          } else if (checker.isRuleMapFunction(node)) {
-            return visitEventBusMap(node);
-          } else if (checker.isNewRule(node)) {
-            return visitRule(node);
-          } else if (checker.isNewEventTransform(node)) {
-            return visitEventTransform(node);
-          } else if (checker.isNewFunctionlessFunction(node)) {
-            return visitFunction(node);
-          } else if (checker.isApiIntegration(node)) {
-            return visitApiIntegration(node);
-          }
-          return node;
-        };
-        // keep processing the children of the updated node.
-        return ts.visitEachChild(visit(), visitor, ctx);
+        if (
+          checker.isApiIntegration(node) ||
+          checker.isAppsyncField(node) ||
+          checker.isAppsyncResolver(node) ||
+          checker.isEventBusWhenFunction(node) ||
+          checker.isNewEventTransform(node) ||
+          checker.isNewFunctionlessFunction(node) ||
+          checker.isNewRule(node) ||
+          checker.isNewStepFunction(node) ||
+          checker.isReflectFunction(node) ||
+          checker.isRuleMapFunction(node)
+        ) {
+          // if this is a Functionless primitive, transform its arguments to wrap functions in their AST form
+          return transform(node);
```

Comment:
What will it take to remove this? tsdoc annotations in the resource?

```ts
class StepFunction {
   /**
     *  @functionless-reflect decl - reflect the callback passed to or defined within the StepFunction resource.
     */ 
   constructor(decl:   )
}
```

Reply:
It will eventually be run on every closure